src/main/java/com/ksm/wstbackoffice/service/CountryService.java

```diff
@@ -23,4 +31,29 @@ public CountryService(CountryRepository countryRepository, CountryMapper country
     public CountryDto findById(String id) {
         return countryMapper.toDto(countryRepository.findById(id).orElse(null));
     }
+
+    public Map<String, Map<String, ?>> findAllBy(Collection<String> nameStartsWithFilters) {
+        Map<String, Map<String, ?>> countriesByNameStartingWith = new HashMap<String, Map<String, ?>>();
+        Map<String, Collection<String>> detailsCountriesByNameStartingWith = new HashMap<String, Collection<String>>();
+        Map<String, Integer> totalCountriesByNameStartingWith = new HashMap<String, Integer>();
+
+        List<CountryDto> countryDtoList = findAll();
+
+        nameStartsWithFilters.stream()
+                .filter(p -> p.trim().length() > 0)
+                .forEach(strFilter -> {
+                    Pattern pattern = Pattern.compile(ValidationConstant.FILTER_NAME_REGEX.concat(strFilter.toLowerCase().trim()));
```

Comment:
Regex should be used very carefully because it reduces the code readability.
Basically: if there is a non-regex operation that does what you want in one step, always go for that.
https://stackoverflow.com/questions/4125377/when-would-it-be-worth-using-regex-in-java